src/main/java/stringCalc/StringParser.java

```diff
@@ -1,37 +1,38 @@
 package stringCalc;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import static stringCalc.ValueValidator.validate;
-
 public class StringParser {
 
-    public static List<String> parse(String customDelimiter, String body) {
+//    public static List<String> parseStringToNumValues(String customDelimiter, String body) {
+//        String expression = getRegularSplitRegex(customDelimiter);
+//        String[] split_nums = body.split(expression);
+//        ValueValidator.validate(List.of(split_nums));
+//        return List.of(split_nums);
+//    }
+
+    public static List<PositiveNumber> parseStringToNumValues(String customDelimiter, String body) {
         String expression = getRegularSplitRegex(customDelimiter);
-        String[] split_nums = body.split(expression);
-        ValueValidator.validate(List.of(split_nums));
-        return List.of(split_nums);
+        String[] splitValues = body.split(expression);
+
+        List<PositiveNumber> splitNums = new ArrayList<>();
+        for (String splitValue : splitValues) {
+            splitNums.add(new PositiveNumber(splitValue));
+        }
```

Comment:
```suggestion
        String[] splitValues = body.split(expression);

        List<PositiveNumber> splitNums = Arrays.stream(splitValues)
                .map(PositiveNumber::new)
                .toList();
```

요렇게 스트림을 사용하는 방법도 있습니다! 스트림을 적절히 사용하면 가독성 개선에 큰 도움이 되니 학습해보시는 것을 추천드려요!